test/test_torch.py

```diff
@@ -3140,10 +3140,17 @@ def torchfn(t1, t2, t3):
                 for first, second, third in [(large, small, small2), (small, large, small2),
                                              (small2, small, large), (small2, large, small)]:
                     if first is None:
-                        break  # ignore last iter when small2 is None
-                    r1 = torchfn(expanded[first], expanded[second], expanded[third])
-                    r2 = torchfn(first, second, third)
-                    self.assertEqual(r1, r2)
+                        return  # ignore last iter when small2 is None
+                    # in-place tensor is not broadcastable; test only guaranteed
+                    # to work by broadcasting other argument(s)
+                    if not hasattr(large_expanded, fn + "_"):
+                        r1 = torchfn(expanded[first], expanded[second], expanded[third])
+                        r2 = torchfn(first, second, third)
+                        self.assertEqual(r1, r2)
+                    else:
+                        r1 = torchfn(expanded[first], expanded[second], expanded[third])
+                        r2 = torchfn(expanded[first], second, third)
+                        self.assertEqual(r1, r2)
```

Comment:
Can we just do it like this (to minimize duplication):
```python
# in-place tensor is not broadcastable; test only guaranteed
# to work by broadcasting other argument(s)
first_maybe_expand = expanded[first] if hasattr(large_expanded, fn + "_") else first
r1 = torchfn(expanded[first], expanded[second], expanded[third])
r2 = torchfn(first_maybe_expand, second, third)
self.assertEqual(r1, r2)
```

Also, it's kind of weird that those ops support this partial broadcasting. I think we should make them fully broadcasting if that's possible.